File: WWF/Hands on labs 3.0/Lab06/Resources/POHelpers.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" as from Windows Workflow Foundation
// Hands on Labs RC
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

namespace ContosoWorkflows
{
    public class POHelpers
    {

        public static void CopyPOItems(POSchema.PO sourcePO, Northwind.PO destinationPO)
        {
            if (sourcePO.Items != null)
            {
                destinationPO.Items = new Northwind.POItem[sourcePO.Items.Length];
                for (int i = 0; i < sourcePO.Items.Length; i++)
                {
                    destinationPO.Items[i] = new Northwind.POItem();
                    destinationPO.Items[i].Sku = sourcePO.Items[i].Sku;
                    destinationPO.Items[i].Price = sourcePO.Items[i].Price;
                    destinationPO.Items[i].Amount = sourcePO.Items[i].Amount;
                }
            }
        }

        public static void CopyPOHistory(POSchema.PO sourcePO, Northwind.PO destinationPO)
        {
            if (sourcePO.History != null)
            {
                destinationPO.History = new Northwind.POStatus[sourcePO.History.Length];
                for (int i = 0; i < sourcePO.History.Length; i++)
                {
                    destinationPO.History[i] = new Northwind.POStatus();
                    destinationPO.History[i].Contact = sourcePO.History[i].Contact;
                    destinationPO.History[i].PoStatus = sourcePO.History[i].PoStatus;
                    destinationPO.History[i].Timestamp = sourcePO.History[i].Timestamp;
                }
            }
        }

        public static void CopyPOHeader(POSchema.PO sourcePO, Northwind.PO destinationPO)
        {
            destinationPO.PONumber = sourcePO.PONumber;
            destinationPO.OrderTotal = sourcePO.OrderTotal;
        }

        public static void CopyPOItems(POSchema.PO sourcePO, Fabrikam.PO destinationPO)
        {
            if (sourcePO.Items != null)
            {
                destinationPO.Items = new Fabrikam.POItem[sourcePO.Items.Length];
                for (int i = 0; i < sourcePO.Items.Length; i++)
                {
                    destinationPO.Items[i] = new Fabrikam.POItem();
                    destinationPO.Items[i].Sku = sourcePO.Items[i].Sku;
                    destinationPO.Items[i].Price = sourcePO.Items[i].Price;
                    destinationPO.Items[i].Amount = sourcePO.Items[i].Amount;
                }
            }
        }

        public static void CopyPOHeader(POSchema.PO sourcePO, Fabrikam.PO destinationPO)
        {
            destinationPO.PONumber = sourcePO.PONumber;
            destinationPO.OrderTotal = sourcePO.OrderTotal;
        }

        public static void CopyPOHistory(POSchema.PO sourcePO, Fabrikam.PO destinationPO)
        {
            if (sourcePO.History != null)
            {
                destinationPO.History = new Fabrikam.POStatus[sourcePO.History.Length];
                for (int i = 0; i < sourcePO.History.Length; i++)
                {
                    destinationPO.History[i] = new Fabrikam.POStatus();
                    destinationPO.History[i].Contact = sourcePO.History[i].Contact;
                    destinationPO.History[i].PoStatus = sourcePO.History[i].PoStatus;
                    destinationPO.History[i].Timestamp = sourcePO.History[i].Timestamp;
                }
            }
        }
    }
}

File: WWF/Hands on labs 3.5/Lab06/VB/after/Exercise1/ContosoWorkflows/POHelpers.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" as from Windows Workflow Foundation
' Hands on Labs RC
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

Imports POSchema

Public Class POHelpers


    

    Public Shared Sub CopyPOHeader(ByVal sourcePO As PO, ByVal destinationPO As Northwind.PO)

        destinationPO.PONumber = sourcePO.PONumber
        destinationPO.OrderTotal = sourcePO.OrderTotal

    End Sub


 
 
    Public Shared Sub CopyPOHistory(ByVal sourcePO As PO, ByVal destinationPO As Northwind.PO)

        If (Not sourcePO.History Is Nothing) Then
            destinationPO.History = New Northwind.POStatus(sourcePO.History.Length - 1) {}
            Dim num1 As Integer = 0
            Do While (num1 < sourcePO.History.Length)
                destinationPO.History(num1) = New Northwind.POStatus
                destinationPO.History(num1).Contact = sourcePO.History(num1).Contact
                destinationPO.History(num1).PoStatus = sourcePO.History(num1).PoStatus
                destinationPO.History(num1).Timestamp = sourcePO.History(num1).Timestamp
                num1 += 1
            Loop
        End If

    End Sub


 
    Public Shared Sub CopyPOItems(ByVal sourcePO As PO, ByVal destinationPO As Northwind.PO)
        If (Not sourcePO.Items Is Nothing) Then
            destinationPO.Items = New Northwind.POItem(sourcePO.Items.Length - 1) {}
            Dim num1 As Integer = 0
            Do While (num1 < sourcePO.Items.Length)
                destinationPO.Items(num1) = New Northwind.POItem
                destinationPO.Items(num1).Sku = sourcePO.Items(num1).Sku
                destinationPO.Items(num1).Price = sourcePO.Items(num1).Price
                destinationPO.Items(num1).Amount = sourcePO.Items(num1).Amount
                num1 += 1
            Loop
        End If
    End Sub


    Public Shared Sub CopyPOHeader(ByVal sourcePO As PO, ByVal destinationPO As Fabrikam.PO)

        destinationPO.PONumber = sourcePO.PONumber
        destinationPO.OrderTotal = sourcePO.OrderTotal

    End Sub




    Public Shared Sub CopyPOHistory(ByVal sourcePO As PO, ByVal destinationPO As Fabrikam.PO)

        If (Not sourcePO.History Is Nothing) Then
            destinationPO.History = New Fabrikam.POStatus(sourcePO.History.Length - 1) {}
            Dim num1 As Integer = 0
            Do While (num1 < sourcePO.History.Length)
                destinationPO.History(num1) = New Fabrikam.POStatus
                destinationPO.History(num1).Contact = sourcePO.History(num1).Contact
                destinationPO.History(num1).PoStatus = sourcePO.History(num1).PoStatus
                destinationPO.History(num1).Timestamp = sourcePO.History(num1).Timestamp
                num1 += 1
            Loop
        End If

    End Sub



    Public Shared Sub CopyPOItems(ByVal sourcePO As PO, ByVal destinationPO As Fabrikam.PO)
        If (Not sourcePO.Items Is Nothing) Then
            destinationPO.Items = New Fabrikam.POItem(sourcePO.Items.Length - 1) {}
            Dim num1 As Integer = 0
            Do While (num1 < sourcePO.Items.Length)
                destinationPO.Items(num1) = New Fabrikam.POItem
                destinationPO.Items(num1).Sku = sourcePO.Items(num1).Sku
                destinationPO.Items(num1).Price = sourcePO.Items(num1).Price
                destinationPO.Items(num1).Amount = sourcePO.Items(num1).Amount
                num1 += 1
            Loop
        End If
    End Sub


End Class



